### PR TITLE
feat: add optional absolute positioning of close button in Tooltip

### PR DESCRIPTION
**Pull request by philipwindeyer**
## Description

Adds an optional `fixedCloseButton` boolean prop to `Tooltip` to optionally absolute position the close button in the top right corner of the Tooltip content.
Also allows space props to be provided to Tooltip (default padding is untouched but can be overridden).

Optionally provided space props also apply to close/x button when  `fixedCloseButton` is true, to match padding/margin of contained context.

Adding this here means using `Tooltip` from roo-ui in `booking-manager` as much as possible instead of writing our own implementation.

💁‍

## Related issues

- https://app.shortcut.com/qantasaccommodation/story/95802/payment-summary-tooltips
- https://www.figma.com/file/YAslMKBAA65t8KR8DoV48y/

💪

### With `fixedCloseButton` option:

Screen Shot 2022-05-25 at 7 01 55 pm

### Regular (i.e. backwards compatible):

Screen Shot 2022-05-25 at 7 01 48 pm

🤨

---

**PR check-list**

🚨 **IMPORTANT:** Please review the [guidelines for contributing](./CONTRIBUTING.md) to this repository first! 👀

- [x] You've requested a review from at least ONE person. Preferably a roo-ui maintainer ([Philip](https://github.com/philipwindeyer), [Eric](https://github.com/eneo5541), [Mike](https://github.com/memcc), [Marcel](https://github.com/Anthropic), [Sam](https://github.com/SamuSan) [Long](https://github.com/KieraDOG)) or a Roo UI working-group human.
  - Note: If your PR has not been reviewed within two days of requesting a review, feel free to seek reviews from other [team members of Qantas Hotels](https://github.com/orgs/hooroo/people).
- [x] Your final merge commit message is in the format `<type>(<scope>): <description>` as per [the contributing guide](https://github.com/hooroo/roo-ui/blob/master/.github/CONTRIBUTING.md#commit-naming)
  - Example: `fix(dependencies): minor version bump to resolve security vulnerability in set-value`

### CODE REVIEW

File: src/components/Tooltip/Tooltip.js [hunk 1]
```diff
@@ -61,42 +64,60 @@ const Tooltip = ({
...
+      ...pick(rest, spaceProps),
+    }),
+  };
+
+  return (
+    <TooltipWrapper
+      pointerDirection={pointerDirection}
+      pointerPosition={pointerPosition}
+      colorMode={colorMode}
+      fixedCloseButton={fixedCloseButton}
```
Comment (philipwindeyer) [inline src/components/Tooltip/Tooltip.js:88]: Open to alternate name suggestions FWIW. This made sense at the time but can appreciate that there's better options

File: src/components/Tooltip/Tooltip.js [hunk 2]
```diff
@@ -61,42 +64,60 @@ const Tooltip = ({
...
+    display: 'inline-block',
+    ml: 'auto',
+    pl: 3,
+    ...(fixedCloseButton && {
+      position: 'absolute',
+      top: 0,
+      right: 0,
+      ...pick(rest, spaceProps),
+    }),
+  };
```
Comment (ajlu) [inline src/components/Tooltip/Tooltip.js:81]: Could possibly change the current `Box` using this to a `styled.div` that can only be customised with `${space}`?

Comment (philipwindeyer) [reply src/components/Tooltip/Tooltip.js]: Done. Much cleaner now 🙂

File: src/components/Tooltip/README.md [hunk 3]
```diff
@@ -18,12 +18,13 @@ export default <Tooltip>Save 45% when using Qantas Points</Tooltip>;
...
-| `colorMode`        | Color mode                          | `One of: light, dark`              | `light` | no        |
-| `onClose`          | Display and behaviour of `x` button | `function`                         | `-`     | no        |
+| Name               | Description                               | Type                               | Default | Required? |
+| :----------------- | :---------------------------------------- | :--------------------------------- | :------ | :-------- |
+| `pointerDirection` | Pointer direction                         | `One of: top, bottom, left, right` | `top`   | no        |
+| `pointerPosition`  | Pointer position                          | `string`                           | `50%`   | no        |
+| `colorMode`        | Color mode                                | `One of: light, dark`              | `light` | no        |
+| `onClose`          | Display and behaviour of `x` button       | `function`                         | `-`     | no        |
+| `fixedCloseButton` | Fixes position of `x` button in top right | `boolean`                          | `false` | no        |
```
[the remark this answers is not in the document]
Comment (philipwindeyer) [reply src/components/Tooltip/README.md:27]: Updated

**Review by ajlu — APPROVED** [review]
🚀

**Pull request merged**